pyrdp/mitm/FastPathMITM.py

```diff
@@ -37,6 +37,34 @@ def onClientPDUReceived(self, pdu: FastPathPDU):
         if self.state.forwardInput:
             self.server.sendPDU(pdu)
 
+        if not self.state.loggedIn:
+            for event in pdu.events:
+                if isinstance(event, FastPathScanCodeEvent):
+                    self.onScanCode(event.scanCode, event.isReleased, event.rawHeaderByte & 2 != 0)
+
     def onServerPDUReceived(self, pdu: FastPathPDU):
         if self.state.forwardOutput:
-            self.client.sendPDU(pdu)
\ No newline at end of file
+            self.client.sendPDU(pdu)
+
+    def onScanCode(self, scanCode: int, isReleased: bool, isExtended: bool):
+        """
+        Handle scan code.
+        """
+        keyName = getKeyName(scanCode, isExtended, self.state.shiftPressed, self.state.capsLockOn)
+
+        if len(keyName) == 1:
+            if not isReleased:
+                self.state.inputBuffer += keyName
+
+        # Left or right shift
+        if scanCode in [0x2A, 0x36]:
```

Comment:
shouldn't these be constants from somewhere?

Reply:
It should, but the only constant we made for that is a dict keycode => friendly name afaik. We should do a constants file from that, we just never did it.